lib/view/weeklyTaskView.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:planner/common/database.dart';
+import 'package:planner/models/task.dart';
+import 'package:planner/view/taskView.dart';
+
+class WeeklyTaskView extends StatefulWidget {
+  const WeeklyTaskView({Key? key}) : super(key: key);
+
+  @override
+  _WeeklyTaskViewState createState() => _WeeklyTaskViewState();
+}
+
+class _WeeklyTaskViewState extends State<WeeklyTaskView> {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+  DatabaseService _db = DatabaseService();
+  List<Task> _allTasks = [];
+  final GlobalKey<RefreshIndicatorState> _refreshIndicatorKey =
+      GlobalKey<RefreshIndicatorState>();
+  @override
+  void initState() {
+    super.initState();
+    fetchData();
+  }
+
+  Future<void> fetchData() async {
+    List<Task> tasks = await fetchWeeklyTask();
+    setState(() {
+      _allTasks = tasks;
+    });
+  }
+
+  Future<List<Task>> fetchWeeklyTask() async {
+    DateTime today = DateTime.now();
+    DateTime start = DateTime(today.year, today.month, today.day);
+    DateTime end = start.add(Duration(days: 7));
+
+    Map<DateTime, List<Task>> activeMap, delayedMap, completedMap;
+
+    // Fetch task maps for the specified week
+    (activeMap, delayedMap, completedMap) = await _db.getTaskMaps(start, end);
+
+    List<Task> allTasks = [
```

Comment:
This bit is the same as lines 38-42 as taskView, can we abstract it somehow to reduce duplication? They'll probably look for that, or at the very least we can write about how we tried to keep it to a minimum in our final essays.

Reply:
hmmmmmmmmmmmmm, let me try to figure it out.

Reply:
Maybe we should make an equivalent function in the db? Since it looks like you want it formatted in a unique way

Reply:
You can use the new db function to get a 1 week